mobly/controllers/android_device.py

```diff
@@ -207,8 +207,15 @@ def get_instances_with_configs(configs):
             raise Error(
                 "Required value 'serial' is missing in AndroidDevice config %s."
                 % c)
-        ad = AndroidDevice(serial)
-        ad.load_config(c)
+        is_required = c.pop("required", True)
```

Comment:
Let's leave everything in with 'get', so that we could check 'required' during start services call as well.